node/node.go

```diff
@@ -613,96 +613,84 @@ func (n *Node) waitRole(ctx context.Context, role string) error {
 	return nil
 }
 
-func (n *Node) runManager(ctx context.Context, securityConfig *ca.SecurityConfig, ready chan struct{}) error {
-	for {
-		if err := n.waitRole(ctx, ca.ManagerRole); err != nil {
-			return err
-		}
-
-		remoteAddr, _ := n.remotes.Select(n.NodeID())
-		m, err := manager.New(&manager.Config{
-			ForceNewCluster: n.config.ForceNewCluster,
-			RemoteAPI: manager.RemoteAddrs{
-				ListenAddr:    n.config.ListenRemoteAPI,
-				AdvertiseAddr: n.config.AdvertiseRemoteAPI,
-			},
-			ControlAPI:     n.config.ListenControlAPI,
-			SecurityConfig: securityConfig,
-			ExternalCAs:    n.config.ExternalCAs,
-			JoinRaft:       remoteAddr.Addr,
-			StateDir:       n.config.StateDir,
-			HeartbeatTick:  n.config.HeartbeatTick,
-			ElectionTick:   n.config.ElectionTick,
-		})
-		if err != nil {
-			return err
-		}
-		done := make(chan struct{})
-		var runErr error
-		go func() {
-			runErr = m.Run(context.Background())
-			close(done)
-		}()
+func (n *Node) startManager(ctx context.Context, securityConfig *ca.SecurityConfig, ready chan struct{}) error {
+	remoteAddr, _ := n.remotes.Select(n.NodeID())
+	m, err := manager.New(&manager.Config{
+		ForceNewCluster: n.config.ForceNewCluster,
+		RemoteAPI: manager.RemoteAddrs{
+			ListenAddr:    n.config.ListenRemoteAPI,
+			AdvertiseAddr: n.config.AdvertiseRemoteAPI,
+		},
+		ControlAPI:     n.config.ListenControlAPI,
+		SecurityConfig: securityConfig,
+		ExternalCAs:    n.config.ExternalCAs,
+		JoinRaft:       remoteAddr.Addr,
+		StateDir:       n.config.StateDir,
+		HeartbeatTick:  n.config.HeartbeatTick,
+		ElectionTick:   n.config.ElectionTick,
+	})
+	if err != nil {
+		return err
+	}
 
+	done := make(chan struct{})
+	var runErr error
+	go func() {
+		runErr = m.Run(context.Background())
+		close(done)
+	}()
+	defer func() {
+		n.setControlSocket(nil)
 		n.Lock()
-		n.manager = m
+		n.manager = nil
```

Comment:
Is there a reason to set `n.manager` to nil when the function returns, or is it just for hygiene? The old loop doesn't seem to do this.